Desktop e-book: Chapter 1 & 2 (#16149)
dotnet/docs: docs/architecture/modernize-desktop/why-modern-desktop-applications.md

> # Why Modern Desktop Applications?

Reviewer: Why there is no metadata for this file ?

## Introduction

### A story of one company

Back in early 2000&#39;s one multinational company started developing a distributed desktop solution to exchange information between different branches of the company and execute optimized operations on centralized units. They have chosen a brand-new framework called Windows Forms for their application development. Over years the project evolved into a mature well tested and time proven application with hundreds of thousands of lines of code. Time passed and .NET Framework 2.0 is no longer the hot new technology. The developers who are working on this application are facing a dilemma. They would like to use the latest stack of technologies in their development and have their application look and &quot;feel&quot; modern. At the same time, they don&#39;t want to throw away the great product they have built over 15 years and rewrite the entire application from scratch.

### Your story

> You might find yourself in the same boat, where you have mature Windows Forms (also known as WinForms) or WPF (Windows Presentation Foundation) applications that have proved their reliability over the years. You probably want to keep using these applications for many more years. At the same time, since those applications were written some time ago, they might be missing capabilities like modern look, performance, integration with new devices and platform features and so on, which gives them a feel of &quot;old tech&quot;. There is another problem that might concern you as a developer. While working on the older .NET Framework versions and maintaining applications that were written a while ago, you might feel like you aren&#39;t learning new technologies and missing on building modern technical skills. If that is your story – this book is for you!

Reviewer 2: Can you replace all the &quot; and &#39; with the actual characters?

Microsoft offered a variety of UI desktop technologies through the years from Win32 introduced in 1995 and to UWP released in 2016.

![](./media/1-1.png)

The most popular technologies for building Windows Desktop according to a survey published by Telerik on April 2016 are Windows Forms, WPF and UWP.

![](./media/1-2.png)

You can develop in any of them using C# and VB.NET, but let&#39;s take a closer look.

### Windows Forms

First released in 2002, Windows Forms is a managed framework and is the oldest, most used, desktop technology built on the Windows GDI engine. It offers a very smooth drag-and-drop experience for developing user interfaces in Visual Studio.  At the same time, Windows Forms relies on the Visual Studio Designer as the main way you develop your UI, so creating visual components from code is not trivial.

- Mature technology with lots of code samples and documentation.
- Very powerful and productive designer. Not so convenient to design UI &quot;from code&quot;.
- Easy and intuitive to learn, thanks to the designer&#39;s drag and drop experience.
- Supported on any Windows version.

> 

Reviewer: ```suggestion
```

| Data | Cloud First Data with Insights. Source of truth in the cloud. Insights to know what happens with your app and how it is performing.  | Local Data.Traditional desktop applications usually need some local data.  |
| Design | Designed for reuse. Reuse in mind between different platforms, front end and back end, running assets in many places as possible.  | Designed for Windows Desktop only  |

As a part of the commitment to provide developers with the best tools to build applications, Microsoft has performed a great effort to bring these concepts, or we can even say platforms, closer together to empower developers with the best of both worlds.To do that, Microsoft has performed a bidirectional effort between the two platforms.

![](./media/1-4.png)

1. Move Desktop Application scenarios into Modern Application platform. The traditional desktop development is still very popular because it addresses certain scenarios really well. It makes sense to take these common desktop scenarios and bring them into the modern desktop platform to make the platform fully capable.

> ![](./media/1-5.png)

Reviewer: ```suggestion
    ![](./media/1-5.png)
```

1. Move Modern Application features into Desktop Applications. For existing desktop apps that need a way to leverage modern capabilities without rewriting from scratch, features from the Modern Application platform are pushed into the Desktop Application.

> ![](./media/1-6.png)

Reviewer: ```suggestion
    ![](./media/1-6.png)
```

## What this guide does not cover

This guide covers a specific subset of scenarios focused on lift and shift scenarios, outlining the way to gain the benefits of modernizing without the effort of rewriting code.

This guide is not about developing modern applications with .NET Core or implementing desktop scenarios into the Modern Application Platform. If focus on how you can leverage your investments on desktop applications while you keep them updated with some of the latest technologies for desktop development.

## Who should use this guide

We wrote this guide for developers and solution architects who want to modernize existing Windows Forms and WPF Desktop Applications to leverage the benefits of .NET Core 3.0 and MSIX installation.

You also might find this guide useful if you are a technical decision maker, such as an enterprise architect or a development lead/director who just wants an overview of the benefits that you can get by updating existing Desktop Apps.

## How to use this guide

> This guide addresses the &quot;why&quot;—why you might want to modernize your existing applications, and the specific benefits you get from using NET Core 3.0 and MSIX to modernize your desktop apps. The content of the guide is designed for architects and technical decision makers who want an overview, but who don&#39;t need to focus on implementation and technical, step-by-step details.

Reviewer: ```suggestion
This guide addresses the "why"—why you might want to modernize your existing applications, and the specific benefits you get from using NET Core 3.0 and MSIX to modernize your desktop apps. The content of the guide is designed for architects and technical decision makers who want an overview, but who don&#39;t need to focus on implementation and technical, step-by-step details.
```

Along the different chapters, sample implementation code snippets and screenshot are provided, with chapter 5 devoted to a showcase a complete migration process for sample applications.

## Sample apps

> To highlight the necessary steps to perform a modernization we will be using a sample application called eShopModernizing. This application has two flavors, Windows Forms and WPF and we will show a step by step process on how to perform the modernization on both of them to .NET Core.

Reviewer: ```suggestion
To highlight the necessary steps to perform a modernization, we will be using a sample application called eShopModernizing. This application has two flavors, Windows Forms and WPF and we will show a step by step process on how to perform the modernization on both of them to .NET Core.
```

> Also, on the GitHub repo for this book, you will find the results of the process, which you can consult with if you decide to follow the step-by-step tutorial.

Reviewer 2: ```suggestion
Also, on the GitHub repository for this book, you'll find the results of the process, which you can consult with if you decide to follow the step-by-step tutorial.
```

## Send us your feedback

> We wrote this guide to help you understand your options for improving and modernizing existing .NET desktop applications. The guide and related sample applications are evolving. We welcome your feedback! If you have comments about how this guide might be more helpful, please send them to dotnet-architecture-ebooks-feedback@service.microsoft.com.

Reviewer 2: ```suggestion
We wrote this guide to help you understand your options for improving and modernizing existing .NET desktop applications. The guide and related sample applications are evolving. We welcome your feedback! If you have comments about how this guide might be more helpful, send them to [dotnet-architecture-ebooks-feedback@service.microsoft.com](mailto:dotnet-architecture-ebooks-feedback@service.microsoft.com).
```
